### PR TITLE
Creates a TVM wheel install

### PR DESCRIPTION
**Pull request by s1113950**
Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,18 +36,30 @@ INCLUDE_FLAGS = -Iinclude -I$(DLPACK_PATH)/include -I$(DMLC_CORE_PATH)/include
...
+# a way to pass in extra args to the various cmake calls
+# for an example see `make build-llvm`
+CMAKE_ARGS =
+
+# a way to pass in extra args to the make calls, ex: running with 8 threads via "MAKE_ARGS='-j8'"
+MAKE_ARGS =
 
 all:
-	@mkdir -p build && cd build && cmake .. && $(MAKE)
+	mkdir -p build && cd build && cmake $(CMAKE_ARGS) .. && $(MAKE) $(MAKE_ARGS)
```
Comment (mshawcroft) [inline Makefile:47]: Looks like @ was inadvertently removed here.

Comment (s1113950) [reply Makefile:47]: good catch! This PR is very much still a PoC so some things might be not in place, but it's good to note this so it gets changed back before any later merge

**Comment by liangfu** [comment]
Just a high-level question, how shall we integrate with LLVM?
The options we have
 * Search for LLVM in default system library, like /usr/lib, /usr/lib/x86_64-linux-gnu/, and /usr/local/lib, /usr/local/llvm/lib
 * Compile llvm as static library and make the TVM wheel independent from the llvm library in the  operating system.

**Comment by mshawcroft** [comment]
> Just a high-level question, how shall we integrate with LLVM?

Hi!  The python world has a series of PEPs that define the standards for portable packages/wheels etc.   PEP513 was the original PEP https://www.python.org/dev/peps/pep-0513/ now superseded by PEP599 https://www.python.org/dev/peps/pep-0599/ 

Those standards define what system .so's a package can assume is present on a system. Any library not defined by those standards should be included within the package itself.  Those PEP's also provide links to various tools to help construct conforming portable packages e.g. the manylinux project that provides dockerized build environments for the construction of conformant wheels and various associated auditing tools.

Short answer, in order to get portable wheels that can be widely distributed and consumed I suspect we will need to package  llvm libraries within the wheel.

**Comment by liangfu** [comment]
> I suspect we will need to package llvm libraries within the wheel.

If that is the case, shall we compile llvm in shared library or static library? We might need a separate RFC to collect ideas from the community.

**Comment by liangfu** [comment]
Hi @s1113950, is there any update lately?

**Comment by s1113950** [comment]
I was told to stop working on this until a later date while I was at OctoML, and then my contract ended. Sorry, I forgot to mention that in a comment on this PR 😞 .
@fhaynes do you know if there's internal 🐙 progress on this? I do know there's a script that we had used that helped build a Python TVM package but it wasn't a wheel specifically. From what I remember, that script might be able to be adapted to this purpose 😄

**Comment by jroesch** [comment]
I think @tqchen was going to pick up this work for the next release, going to close for now, feel free to re-open if otherwise.

**Pull request closed without merging**